packages/runtime-dom/__tests__/customElement.spec.ts

```diff
@@ -243,6 +245,28 @@ describe('defineCustomElement', () => {
         detail: [1]
       })
     })
+
+    test('emit from within async component wrapper', async () => {
+      const E = defineCustomElement(
+        defineAsyncComponent(
+          () => new Promise<typeof CompDef>(res => res(CompDef as any))
+        )
+      )
+      customElements.define('my-async-el-emits', E)
+      container.innerHTML = `<my-async-el-emits></my-async-el-emits>`
+      const e = container.childNodes[0] as VueElement
+      const spy = jest.fn()
+      e.addEventListener('my-click', spy)
+      // this feels brittle but seems necessary to reach the node in the DOM.
+      await nextTick()
+      await nextTick()
+      await nextTick()
```

Comment:
You can replace these `await nextTick()` calls with [`customElements.whenDefined()`](https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry/whenDefined), which is a `Promise` that resolves when the element is *defined*, where  the element is connected and the shadow root is available.

I locally verified this with your changeset.

```suggestion
      await customElements.whenDefined('my-async-el-emits')
```

Reply:
TIL, thanks! 

I will play with this. Think it still needs one nextTick to wait for Vue's update cycle to finish after the custom element has connected, but that would be normal.